crates/via-router/src/lib.rs

```diff
@@ -49,11 +47,14 @@ impl<T> Router<T> {
     }
 
     pub fn visit(&self, path: &str) -> Vec<Result<Found, VisitError>> {
-        let mut segments = SmallVec::new();
+        let mut segments = Vec::with_capacity(8);
+        let mut results = Vec::with_capacity(8);
```

Comment:
The vectors above are used to store path segments and results for matching routes temporarily. The vectors are used for only a brief duration of time. Therefore, I don't think we have to be concerned about requesting a predictable amount of memory from the heap. I'm curious how the community feels about this.